crates/sui-graphql-rpc/src/context_data/db_backend.rs

```diff
@@ -16,6 +16,30 @@ use diesel::{
     sql_types::Text,
 };
 
+/// An enum that determines whether to select rows that are greater than or less than the cursor.
+#[derive(Clone, Copy, PartialEq)]
+pub(crate) enum CursorBound<T> {
+    Gt(T),
+    Lt(T),
+}
+
+/// Controls whether the query is in ascending or descending order.
+#[allow(dead_code)]
+#[derive(Clone, Copy, PartialEq)]
+pub(crate) enum SortOrder {
+    Asc,
+    Desc,
+}
+
+impl SortOrder {
+    pub fn invert(&self) -> Self {
+        match self {
+            SortOrder::Asc => SortOrder::Desc,
+            SortOrder::Desc => SortOrder::Asc,
+        }
+    }
+}
```

Comment:
I recall we talked before (in a previous PR, and during the meeting this week) about not including the ability to order cursors as part of the MVP -- please stop adding more stuff in service of that!

Reply:
So I think the various combinations of `before/after/first/last` may require the db interface to change the order of the results in memory, but I don't think it requires having to provide control over that -- this feature is only necessary if you want to support allowing the GraphQL layer to dictate the order.

Reply:
the various combinations of those cursors + limits may also require the db interface to specify sorting by asc or desc in addition to changing the order of results in memory. 

what i can do is remove usage of SortOrder in db_data_provider, would that satisfy the request?

I think we'll need to support dictating the order pretty soon since apps like the explorer return checkpoints, epochs, and tx blocks in descending order from most recent to least recent

Reply:
Yep, you may need to change the order requested of the DB in conjunction with reversing in memory, and that complexity is fine, it is indeed the logic for controlling the sort order for a given request (in a way that is correct for cursor connections) that I want to avoid.